src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -915,18 +915,32 @@ protected virtual void ValidateConditions(SamlSecurityToken securityToken, Token
                 throw LogArgumentNullException(nameof(securityToken.Assertion));
 
             if (securityToken.Assertion.Conditions == null || securityToken.Assertion.Conditions.Conditions.Count() == 0)
+            {
+                if (validationParameters.RequireAudience)
+                    throw LogExceptionMessage(new SamlSecurityTokenException(LogMessages.IDX11401));
+               
                 return;
+            }
 
-            Validators.ValidateLifetime(securityToken.Assertion.Conditions.NotBefore, securityToken.Assertion.Conditions.NotOnOrAfter, securityToken, validationParameters);
+            ValidateLifetime(securityToken.Assertion.Conditions.NotBefore, securityToken.Assertion.Conditions.NotOnOrAfter, securityToken, validationParameters);
 
+            var foundAudienceRestriction = false;
             if (securityToken.Assertion.Conditions.Conditions.ElementAt(0) is SamlAudienceRestrictionCondition)
             {
                 foreach (var condition in securityToken.Assertion.Conditions.Conditions)
```

Comment:
this code is not changed, wondering why this needs to be different from how the loop is setup for SAML2

Reply:
because Saml and Saml2 had to be different

Reply:
makes sense, however the type is not necessarily "validated" it's assumed it's of type Saml2AudienceRestrictionCondition ? (only the first element type is validated). probably missing something here.

Reply:
@henrik-me I think you are correct, the logic requires the first condition to be a AudienceRestriction.
The first clause with ElementAt(0) should be removed.

Reply:
I removed the clause requiring the first condition to be an AudienceRestrictionCondition from the SamlSecurityTokenHandler.  I also added some extra null checks for the Conditions property (for both Saml1 and Saml2) as if it's null we will currently throw a null reference exception. Based on this [spec](http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf) it seems that the Conditions element is optional, so we probably shouldn't be throwing.